src/Instrumentation/ExtRdKafka/tests/Integration/ExtRdKafkaInstrumentationTest.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Instrumentation\ExtRdKafka\tests\Integration;
+
+use ArrayObject;
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\Configurator;
+use OpenTelemetry\API\Trace\Propagation\TraceContextPropagator;
+use OpenTelemetry\Context\ScopeInterface;
+use OpenTelemetry\SDK\Trace\ImmutableSpan;
+use OpenTelemetry\SDK\Trace\SpanExporter\InMemoryExporter;
+use OpenTelemetry\SDK\Trace\SpanProcessor\SimpleSpanProcessor;
+use OpenTelemetry\SDK\Trace\TracerProvider;
+use PHPUnit\Framework\TestCase;
+use RdKafka\Conf;
+use RdKafka\KafkaConsumer;
+use RdKafka\Message;
+use RdKafka\Producer;
+
+class ExtRdKafkaInstrumentationTest extends TestCase
+{
+    private const STANDARD_TOPIC_NAME = 'test';
+    private ScopeInterface $scope;
+    private ArrayObject $storage;
+    private TracerProvider $tracerProvider;
+
+    public function setUp(): void
+    {
+        $this->storage = new ArrayObject();
+        $this->tracerProvider = new TracerProvider(
+            new SimpleSpanProcessor(
+                new InMemoryExporter($this->storage)
+            )
+        );
+        $this->scope = Configurator::create()
+            ->withTracerProvider($this->tracerProvider)
+            ->withPropagator(new TraceContextPropagator())
+            ->activate();
+    }
+
+    public function tearDown(): void
+    {
+        $this->scope->detach();
+    }
+
+    public static function tearDownAfterClass(): void
```

Comment:
it should probably do this after every test. I'm not sure if we're doing it in this repo yet, but over in the main repo we run all tests in random order to shake out any inter-test dependencies.

Reply:
Good point- toyed with the idea but was worried about the overhead (these tests already take a little while to run). After a test, it's not really noticeable speed-wise. Have changed it